### ISSUE TITLE
Could not create DeckLink iterator.

### ISSUE THREAD

**Issue by alexhabgood** [amiaopensource/vrecord#734]
Hiya,

I recently did a brew update/upgrade on an Ubuntu 20.04.5 LTS workstation, which has vrecord installed. It was working fine before the update, but now vrecord fails to find my decklink duo card. It gives the error:

>libDeckLinkAPI.so: cannot open shared object file: No such file or directory
>[decklink @ 0x32d6b00] Could not create DeckLink iterator. Make sure you have DeckLink drivers 10.11.2 or newer installed.

I've got Desktop Video 12.2.2 installed. Running:
>dkms status -m blackmagic-io -k `uname -r`

Returns:
> blackmagic-io, 12.2.2a6, 5.14.0-1054-oem, x86_64: installed

Any ideas on how to fix this?
Thanks in advance :)

**Comment by privatezero** [comment]
Hi @alexhabgood, I don't have an Ubuntu 20.04 machine that I can test with an actual Blackmagic device, so I can't try to recreate your issue exactly, so kind of guessing as to possible fixes. Have you tried reinstalling ffmpeg decklink?

**Comment by alexhabgood** [comment]
Hi @privatezero 

Thanks for responding, greatly appreciated.

I've done some updates to my system. Now on Ubuntu 22.04.1 LTS.
Blackmagic drivers have also been updated to the latest 12.4.1.
Have tested Media Express and that's working fine.

Subsequently I tried brew reinstall ffmpegdecklink and that goes through fine without any errors. However, vrecord still doesn't seem to be finding my decklink card. 

Now when I boot vrecord -e. It returns:

> /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl: error while loading shared libraries: libraw1394.so.11: cannot open shared object file: No such file or directory

I tried reinstalling libraw1394-dev. But that didn't change anything.

**Comment by privatezero** [comment]
Hi @alexhabgood  - that library is related to the DV capture functionality so that at least suggests a bit of an avenue for trouble shooting. As a bit of triage, does reinstalling ffmpegdecklink without enabling `--with-iec61883` result in a more functional build that at least can work with Blackmagic?

**Comment by alexhabgood** [comment]
I reinstalled ffmpegdecklink without iec61883. See the attached log.

When I launch vrecord it now gives me the error:
> /home/linuxbrew/.linuxbrew/opt/ffmpegdecklink/bin/ffmpeg-dl: error while loading shared libraries: libasound.so.2: cannot open shared object file: No such file or directory

I tried reinstalling libasound2, but that didn't change anything. Decklink card still isn't showing in the settings menu either.

I also ran:
> ldconfig -p 
But that seems to be fine (also attached).  

[ldconfig.txt](https://github.com/amiaopensource/vrecord/files/10083702/ldconfig.txt)

[ffmpegdecklink_install.txt](https://github.com/amiaopensource/vrecord/files/10083601/ffmpegdecklink_install.txt)

**Comment by alexhabgood** [comment]
Hi @privatezero

Tried running pulseaudio and discovered it was also having dificulty finding shared libraries. Thought it might be something fundamentally broken in my homebrew install, so I completely uninstalled homebrew and all packages. Then tried reinstalling vrecord and managed to get it working. When I did the reinstall I did experience issue #731, but bypassed it by just not installing deckcontrol. 

Not entirely sure what went wrong, but will close this now :)
